Sources/Widgets/Widgets3D/LineWidget/behavior.js

```diff
@@ -128,60 +136,59 @@ export default function widgetBehavior(publicAPI, model) {
     );
   }
 
-  function orientFirstHandleBeforeSecondHandlePlacement(callData) {
-    const handle1Origin = model.widgetState.getHandle1().getOrigin();
+  function computeMousePosition(p1, callData) {
     const worldMousePos = publicAPI.computeWorldToDisplay(
       model.renderer,
-      handle1Origin[0],
-      handle1Origin[1],
-      handle1Origin[2]
+      p1[0],
+      p1[1],
+      p1[2]
     );
     const mousePos = publicAPI.computeDisplayToWorld(
       model.renderer,
       callData.position.x,
       callData.position.y,
       worldMousePos[2]
     );
-    vtkMath.subtract(
-      model.widgetState.getHandle1().getOrigin(),
-      mousePos,
-      direction
-    );
+    return mousePos;
   }
 
-  function orientHandlesWithCompleteWidget(modifier) {
+  function orientFirstHandleBeforeSecondHandlePlacement(callData) {
     const handle1Origin = model.widgetState.getHandle1().getOrigin();
-    const handle2Origin = model.widgetState.getHandle2().getOrigin();
-    vtkMath.subtract(handle1Origin, handle2Origin, direction);
-    vtkMath.multiplyScalar(direction, modifier);
+    const mousePos = computeMousePosition(handle1Origin, callData);
+    direction = computeDirectionWithTwoPoints(handle1Origin, mousePos);
```

Comment:
return the return value of "computeDirectionWithTwoPoints" and use the returned value instead of using "global" variable" `direction`